src/almanack/processing/repository_processing.py

```diff
@@ -6,59 +6,62 @@
 import shutil
 import tempfile
 from datetime import datetime, timezone
+from typing import Optional, Tuple
 
 import pygit2
 
-from .data_management import save_entropy_to_json
-from .entropy import aggregate_entropy_calculation
+from .entropy_calculations import calculate_aggregate_entropy
 from .git_operations import clone_repository, get_commits, get_edited_files
 
 
-def process_entire_repo(repo_path: str, output_path: str) -> None:
+def process_entire_repo(repo_path: str) -> None:
     """
     Processes a repository path to calculate entropy.
 
     Args:
         repo_path (str): The local path to the Git repository.
-        output_path (str): Path to the output JSON file.
 
     Returns:
-        None: Saves the entropy data to a JSON file.
+        Dict: Key Value pair containing repository path and normalized entropy
     """
     try:
+        # Convert repo_path to an absolute path and initialize the repository
         repo_path = pathlib.Path(repo_path).resolve()
         repo = pygit2.Repository(str(repo_path))
 
+        # Retrieve the list of commits from the repository
         commits = get_commits(repo)
         most_recent_commit = commits[0]
         first_commit = commits[-1]
 
+        # Get a list of files that have been edited between the first and most recent commit
         file_names = get_edited_files(repo, first_commit, most_recent_commit)
 
-        normalized_total_entropy = aggregate_entropy_calculation(
+        # Calculate the normalized total entropy for the repository
+        normalized_total_entropy = calculate_aggregate_entropy(
             repo_path,
             str(first_commit.id),
             str(most_recent_commit.id),
             file_names,
         )
 
-        # Implement soon
-        # file_level_entropy = calculate_normalized_entropy(
-        #     repo_path,
-        #     str(first_commit.id),
-        #     str(most_recent_commit.id),
-        #     file_names,
-        # )
+        # Prepare the data structure
+        data = {
+            "repo_path": str(repo_path),
+            "total_normalized_entropy": normalized_total_entropy,
+        }
 
-        save_entropy_to_json(str(repo_path), normalized_total_entropy, output_path)
+        return data
 
     except Exception as e:
-        print(f"Error processing repository {repo_path}: {e}")
+        # If processing fails, return an error dictionary
+        error_data = {"repo_path": str(repo_path), "error": str(e)}
+        return error_data
```

Comment:
Consider returning the data directly instead of forming a named variable which is used only once for return.